modules/andes-core/client/src/main/java/org/wso2/andes/client/XAConnectionImpl.java

```diff
@@ -43,29 +46,47 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(XAConnectionImpl.class);
 
     /**
+     * After this delay(in seconds), the connection will be closed even if a commit or rollback is not received. This
+     * is done to avoid stale connections to broker.
+     */
+    private static final int CONNECTION_CLOSE_TIMEOUT = 60;
+
+    /**
      * Keep track of active XA sessions
      */
     private final ArrayList<XASession_9_1> xaSessions = new ArrayList<>();
 
     /**
+     * Executor service used to schedule connection close task.
+     */
+    private final ScheduledExecutorService scheduledExecutor;
+
+    /**
      * Indicate if the connection.close is called before
      */
     private boolean connectionCloseSignaled = false;
 
+    /**
+     * Hold the future object of the scheduled close task
+     */
+    private ScheduledFuture<?> connectionCloseFuture;
+
     //-- constructor
     /**
      * Create a XAConnection from a connectionURL
      */
-    XAConnectionImpl(ConnectionURL connectionURL, SSLConfiguration sslConfig) throws AMQException {
+    XAConnectionImpl(ConnectionURL connectionURL, SSLConfiguration sslConfig,
+            ScheduledExecutorService scheduledExecutor) throws AMQException {
         super(connectionURL, sslConfig);
+        this.scheduledExecutor = scheduledExecutor;
```

Comment:
All the usages of scheduled executor are within XAConnectionImpl. So why do we need to pass it as a parameter to the constructor?